assets/model_monitoring/components/src/model_monitor_metric_publisher/run.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Entry script for Model Monitor Azure Monitor Metric Publisher Component."""
+
+import argparse
+import os
+
+from pyspark.sql import Row
+from typing import List
+
+from azure_monitor_metric_publisher import extract_location_from_aml_endpoint_str, publish_metric
+from shared_utilities.datetime_utils import parse_datetime_from_string
+from shared_utilities.io_utils import read_mltable_in_spark
+from shared_utilities.log_utils import log_error
+
+
+def run():
+    """Publish Azure monitor metrics."""
+    arg_parser = argparse.ArgumentParser()
+    arg_parser.add_argument("--signal_metrics", type=str)
+    arg_parser.add_argument("--monitor_name", type=str)
+    arg_parser.add_argument("--signal_name", type=str)
+    arg_parser.add_argument("--data_window_start", type=str)
+    arg_parser.add_argument("--data_window_end", type=str)
+    args = arg_parser.parse_args()
+
+    metrics: List[Row] = read_mltable_in_spark(args.signal_metrics).collect()
+
+    try:
+        azureml_service_endpoint = os.environ['AZUREML_SERVICE_ENDPOINT']
+        workspace_resource_id = os.environ['AZUREML_WORKSPACE_SCOPE']
+    except KeyError as err:
+        log_error(f"Failed to find a required environment variable, error: {str(err)}.")
+        raise
+
+    location = extract_location_from_aml_endpoint_str(azureml_service_endpoint)
```

Comment:
Is it possible to Get on WorkspaceResourceId to get the location info instead?

Reply:
No, I don't think so. `WorkspaceResourceId` doesn't contain location info, which is like, `/subscriptions/{sub_id}/resourceGroups/{rg_name}/providers/Microsoft.MachineLearningServices/workspaces/{ws_name}`.